service-base/src/main/java/org/eclipse/hono/service/auth/device/SubjectDnCredentials.java

```diff
@@ -30,6 +30,8 @@
  */
 public class SubjectDnCredentials extends AbstractDeviceCredentials {
 
+    private byte[] clientCertificate = null;
```

Comment:
I would rather have a generic way to add custom properties to any type of credentials. This is actually what we promised with the Credentials API but never stood up to. We originally introduced that feature in teh API to allow the MQTT adapter to include the MQTT client-id in the get Credentials request but never implemented it.
This seems to be a good opportunity to introduce this in a generic way. I can imagine that the `AbstractDeviceCredenials` class supports a `JsonObject` typed property that can be set in the constructor. WDYT?

Reply:
I like the idea. If you want to have the custom properties in any type of credentials, shouldn't it be added to the interface `DeviceCredentials`?

Reply:
I guess so. Just remember to introduce it in a backwards compatible way ...